source/run-command/run-command.rst

```diff
@@ -0,0 +1,185 @@
+
+* See Driver Sessions' section on `Sending the session ID to the server on all commands <https://github.com/mongodb/specifications/blob/master/source/sessions/driver-sessions.rst#sending-the-session-id-to-the-server-on-all-commands>`_
+
+Transactions
+""""""""""""
+
+If RunCommand is used within a Transaction the read preference MUST be sourced from the Transaction's options.
+The command sent to the server MUST include the transaction specific fields, summarized as follows:
+
+* If ``runCommand`` is executing within a transaction:
+
+  * ``autocommit`` - The autocommit flag MUST be set to false.
+  * ``txnNumber`` - MUST be set.
+
+* If ``runCommand`` is the first operation of the transaction:
+
+  * ``startTransaction`` - MUST be set to true.
+  * ``readConcern`` - MUST be set to the Transaction's readConcern setting if it is NOT the default.
+
+* See Transactions' section on `Generic RunCommand helper within a transaction <https://github.com/mongodb/specifications/blob/master/source/transactions/transactions.rst#generic-runcommand-helper-within-a-transaction>`_
+
+ReadConcern and WriteConcern
+""""""""""""""""""""""""""""
+
+A RunCommand API MUST NOT support Read Concern and Write Concern options.
+Additionally, unless within a transaction, the command sent MUST NOT have any Read Concern or Write Concern fields applied that may be inherited from client, database, or collection options.
+
+* See Read Concern's section on `Generic Command Method <https://github.com/mongodb/specifications/blob/master/source/read-write-concern/read-write-concern.rst#generic-command-method>`_
+* See Write Concern's section on `Generic Command Method <https://github.com/mongodb/specifications/blob/master/source/read-write-concern/read-write-concern.rst#generic-command-method-1>`_
+
+Retryability
+""""""""""""
+
+All commands passed to runCommand are considered non-retryable reads.
+Drivers MUST NOT inspect the command to determine if it is a write and MUST NOT attach a ``txnNumber``.
+
+* See Retryable Reads' section on `Unsupported Read Operations <https://github.com/mongodb/specifications/blob/master/source/retryable-reads/retryable-reads.rst#unsupported-read-operations>`_
+* See Retryable Writes' section on `Behavioral Changes for Write Commands <https://github.com/mongodb/specifications/blob/master/source/retryable-writes/retryable-writes.rst#behavioral-changes-for-write-commands>`_
+
+Stable API
+""""""""""
+
+The command sent MUST attach stable API fields as configured on the MongoClient.
+
+* See Stable API's section on `Generic Command Helper Behaviour <https://github.com/mongodb/specifications/blob/master/source/versioned-api/versioned-api.rst#generic-command-helper>`_
+
+Client Side Operations Timeout
+""""""""""""""""""""""""""""""
+
+A driver's runCommand API MUST provide an optional timeoutMS option to support client side operations timeout.
+Drivers MUST NOT attempt to check the command document for the presence of a ``maxTimeMS`` field.
```

Comment:
This language might be something we can use earlier when talking about not modifying existing ``readConcern`` or ``writeConcern`` fields on the command document.

Likewise, should we add similar text for the section talking about ``lsid`` setting? IIRC, drivers are supposed to append the field regardless of whether it exists on the user-supplied document or not. And depending on the driver and BSON representation I think that could result in duplicate fields or overwriting the existing value -- which we don't really care about since the user is probably doing something they shouldn't.

Reply:
Added notes to session, RW concern. And improved the detail in the implementation details section about modification to highlight append vs overwrite